sa/storage-authority.go

```diff
@@ -213,6 +213,36 @@ func (ssa *SQLStorageAuthority) GetAuthorization(id string) (authz core.Authoriz
 	return
 }
 
+// GetAuthorizationsByDomain obtains all authorizations for a domain name
+func (ssa *SQLStorageAuthority) GetAuthorizationsByDomain(domain core.AcmeIdentifier) ([]core.Authorization, error) {
```

Comment:
Yeah, I think RevokeAuthorizationsForDomain in the SA sounds better, for the reasons you say.